ws/client/src/lib.rs

```diff
@@ -0,0 +1,37 @@
+use failure::{Error, format_err};
+use futures::prelude::*;
+use futures::sync::mpsc;
+use websocket::{ClientBuilder, OwnedMessage};
+use jsonrpc_core::client::{RpcClient, RpcChannel, RpcError};
+
+pub fn connect<T>(url: &str) -> Result<impl Future<Item=T, Error=RpcError>, Error>
+where
+	T: From<RpcChannel>,
+{
+	let client = ClientBuilder::new(url)?
+		.async_connect(None)
+		.map(|(client, _)| {
+			let (sink, stream) = client
+				.map_err(|err| RpcError::Other(err.into()))
+				.sink_map_err(|err| RpcError::Other(err.into()))
+				.with(|data| -> Result<_, RpcError> {
+					Ok(OwnedMessage::Text(data))
+				})
+				.and_then(|recv| {
+					match recv {
+						OwnedMessage::Text(data) => Ok(data),
+						// TODO: pings and pongs and close messages need to be
+						//       answered; for now we produce an error when that
+						//       happens
+						_ => Err(RpcError::Other(format_err!("unimplemented"))),
+					}
+				}).split();
+			let (sender, receiver) = mpsc::channel(0);
```

Comment:
I guess unbufferred channel here is fine, but that's something to mention in docs.